sh_scrapy/extension.py

```diff
@@ -48,13 +49,24 @@ def spider_closed(self, spider, reason):
         self.pipe_writer.set_outcome(reason)
 
 
+_DOWNLOADER_MIDDLEWARE_WARNING = """\
+sh_scrapy.extension.HubstorageMiddleware functionality is now split between two new middlewares:
+
+- sh_scrapy.middlewares.HubstorageDownloaderMiddleware
+- sh_scrapy.middlewares.HubstorageSpiderMiddleware
+
+Please migrate to new middlewares.
+"""
+
+
 class HubstorageMiddleware(object):
 
     def __init__(self):
+        warnings.warn(_DOWNLOADER_MIDDLEWARE_WARNING)
```

Comment:
btw, do you agree that users that extend old class (if any) are also disabling old middldeware manually and enabling their custom extended middleware? If you do, consider that custom middleware and new middleware are unlikely to be compatible.